code/modules/clothing/spacesuits/hardsuit.dm

```diff
@@ -428,3 +428,102 @@
+		var/datum/effect_system/spark_spread/s = new
+		s.set_up(2, 1, src)
+		s.start()
+		owner.visible_message("The [attack_text] sparks harmlessly off [owner]'s shields!")
+		current_charges--
+		recharge_cooldown = world.time + recharge_delay
+		if(current_charges <= 0)
+			owner.visible_message("[owner]'s shield overloads!")
+			SHIELD_STATE = "broken"
+			owner.update_inv_wear_suit()
+		return 1
+	return 0
+
+
+
+/obj/item/clothing/suit/space/hardsuit/shielded/New()
+	..()
+	SSobj.processing |= src
+
+/obj/item/clothing/suit/space/hardsuit/shielded/Destroy()
+	SSobj.processing.Remove(src)
+	return ..()
+
+/obj/item/clothing/suit/space/hardsuit/shielded/process()
+	if(world.time > recharge_cooldown && current_charges < max_charges)
+		current_charges = Clamp((current_charges + recharge_rate), 0, max_charges)
+		playsound(loc, 'sound/magic/Charge.ogg', 50, 1)
+		if(current_charges == max_charges)
+			playsound(loc, 'sound/machines/ding.ogg', 50, 1)
+		SHIELD_STATE = "[SHIELD_ON]"
+		if(istype(loc, /mob/living/carbon/human))
+			var/mob/living/carbon/human/C = loc
+			C.update_inv_wear_suit()
+
+
+
+/obj/item/clothing/suit/space/hardsuit/shielded/worn_overlays(isinhands)
+    . = list()
+    if(!isinhands)
+        . += image(icon = 'icons/effects/effects.dmi', icon_state = "[SHIELD_STATE]")
+
+
+///////////////Capture the Flag////////////////////
+
+/obj/item/clothing/suit/space/hardsuit/shielded/ctf
+	name = "red shielded hardsuit"
+	desc = "Standard issue hardsuit for playing capture the flag."
+	icon_state = "hardsuit0-ert_security"
+	item_state = "hardsuit0-ert_security"
+	item_color = "ert_security"
+	flags = STOPSPRESSUREDMAGE | THICKMATERIAL | NODROP //Dont want people changing into the other teams gear
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/ert/ctf
+	armor = list(melee = 0, bullet = 30, laser = 30, energy = 30, bomb = 50, bio = 100, rad = 100)
+	slowdown = 0
+
+/obj/item/clothing/suit/space/hardsuit/shielded/ctf/blue
+	name = "blue shielded hardsuit"
+	desc = "Standard issue hardsuit for playing capture the flag."
+	icon_state = "ert_command"
+	item_state = "ert_command"
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/ert/ctf/blue
+	armor = list(melee = 0, bullet = 30, laser = 30, energy = 30, bomb = 50, bio = 100, rad = 100)
+	slowdown = 0
+
+
+/obj/item/clothing/head/helmet/space/hardsuit/ert/ctf
+	name = "shielded hardsuit helmet"
+	desc = "Standard issue hardsuit helmet for playing capture the flag."
+	icon_state = "hardsuit0-ert_security"
+	item_state = "hardsuit0-ert_security"
+	item_color = "ert_security"
+	armor = list(melee = 0, bullet = 30, laser = 30, energy = 30, bomb = 50, bio = 100, rad = 100)
+
+/obj/item/clothing/head/helmet/space/hardsuit/ert/ctf/blue
+	name = "shielded hardsuit helmet"
+	desc = "Standard issue hardsuit helmet for playing capture the flag."
+	icon_state = "hardsuit0-ert_commander"
+	item_state = "hardsuit0-ert_commander"
+	item_color = "ert_commander"
```

Comment:
Because they have the same icons as ERT suits and ERT suits hold all the guns and stuff I want these to hold anyway